packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -1,258 +1,259 @@
     const formatPath = (path: string[]) => ({
       params: {
         path,
       },
     });
 
-    return this.fetch([locale], rootItemPath, formatPath, formatSearchQuery);
+    return this.fetchSitemap([locale], rootItemPath, formatPath);
   }
 
   /**
    * Fetch sitemap which could be used for generation of static pages using SSG mode
-   * @param {string[]} locales locales which application supports
+   * @param {string[]} locales which application supports
    * @param {string} rootItemPath root item path
-   * @param {Function} [formatSearchQuery] override default search query
-   *
-   * @default
-   * Search query
-   * search(
-   *    where: {
-   *      AND:[
-   *        {
-   *           name:"_path",
-   *          value:"${rootItemId.toLowerCase()}"
-   *        },
-   *        {
-   *          name:"_language",
-   *          value:"${locale}"
-   *        },
-   *        {
-   *          name:"_hasLayout",
-   *          value :"true"
-   *        }
-   *      ]
-   *    }
-   *  )
+   * @throws {RangeError} if a valid root item ID cannot be determined from the specified root item path.
+   * @throws {RangeError} if the specified locale is not valid.
    */
-  async fetchSSGSitemap(
-    locales: string[],
-    rootItemPath: string,
-    formatSearchQuery?: (rootItemId: string, locale: string) => string
-  ): Promise<StaticPath[]> {
+  async fetchSSGSitemap(locales: string[], rootItemPath: string): Promise<StaticPath[]> {
     const formatPath = (path: string[], locale: string) => ({
       params: {
         path,
       },
       locale,
     });
 
-    return this.fetch(locales, rootItemPath, formatPath, formatSearchQuery);
+    return this.fetchSitemap(locales, rootItemPath, formatPath);
   }
 
   /**
-   * Internal fetch
-   * @param {string[]} locales
+   * Gets the list of site pages.
+   * @param {string[]} languages
    * @param {string} rootItemPath
    * @param {Function} formatStaticPath
-   * @param {Function} [formatSearchQuery]
+   * @default Search query
+   * query SitePageQuery(
+   *   $rootItemId: String!,
+   *   $language: String!,
+   *   $pageSize: Int = 10,
+   *   $after: String
+   * ) {
+   *   search(
+   *     where: {
+   *       AND:[
+   *         { name: "_path",      value: $rootItemId },
+   *         { name: "_language",  value: $language   },
+   *         { name: "_hasLayout", value: "true"      }
+   *       ]
+   *     }
+   *     first: $pageSize
+   *     after: $after
+   *   ) {
+   *     total
+   *     pageInfo {
+   *       endCursor
+   *       hasNext
+   *     }
+   *     results {
+   *       url {
+   *         path
+   *       }
+   *     }
+   *   }
+   * }
+   * @throws {RangeError} if a valid root item ID cannot be determined from the specified root item path.
+   * @throws {RangeError} if the specified language list is not valid.
    */
-  private async fetch(
-    locales: string[],
+  async fetchSitemap(
```

Comment:
@anastasiya29 Should be private

Reply:
This was intentionally made public because this is the new way how devs can customize the query. Before they had to pass in a 'formatQuery' function; now they have to inherit from this class and override this function.

Reply:
@anastasiya29 Let's imagine that I extended this class, and in this case I don't have access to `this.getRootItemId` - since it's private, same for `this.fetch`. Also in this case it's required to define `const client = new GraphQLRequestClient(this.options.endpoint)`. But what if I just want to customize search query? In our case I need to duplicate logic. Is it not overcomplicated?

Reply:
@anastasiya29 If this is only intended to be overridden in subclasses, we should probably make this `protected`. Having 3 public "fetch" methods is going to be confusing.

Reply:
Fair point. Done.